store/src/java/com/zimbra/cs/zimlet/ZimletUtil.java

```diff
@@ -195,24 +195,57 @@ public static ZimletPresence getAvailableZimlets(Account acct) throws ServiceExc
         if (domain != null) {
             String[] domainZimlets = domain.getMultiAttr(Provisioning.A_zimbraZimletDomainAvailableZimlets);
             for (String zimletWithPrefix : domainZimlets) {
-                availZimlets.put(zimletWithPrefix);
+                if (isNotZextrasZimlet(zimletWithPrefix)) {
+                    availZimlets.put(zimletWithPrefix);
+                }
             }
         }
 
         String[] acctCosZimlets = acct.getMultiAttr(Provisioning.A_zimbraZimletAvailableZimlets);
         for (String zimletWithPrefix : acctCosZimlets) {
-            availZimlets.put(zimletWithPrefix);
+            if (isNotZextrasZimlet(zimletWithPrefix)) {
+                availZimlets.put(zimletWithPrefix);
+            }
         }
 
         return availZimlets;
     }
+    
+    /**
+     * this method is also check whether zimletName is having prefix like +,- and !
+     * and this method is used to find out whether zimletName is related to remove
+     * ZEXTRAS list or not if zimletName is not in remove ZEXTRAS list then it
+     * returns true else it returns false
+     *
+     * @param zimletName
+     * @return boolean
+     */
+    private static boolean isNotZextrasZimlet(String zimletName) {
```

Comment:
What if we make it `isZextrasZimlet()` and return accordingly?

Reply:
renamed